app/javascript/components/stripe/StripeForm.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { useCallback, useState } from 'react'
+import { Icon } from '../common'
+import { sendRequest } from '../../utils/send-request'
+import { loadStripe, StripeCardElementChangeEvent } from '@stripe/stripe-js'
+import {
+  Elements,
+  CardElement,
+  useStripe,
+  useElements,
+} from '@stripe/react-stripe-js'
+import { fetchJSON } from '../../utils/fetch-json'
+
+type PaymentIntent = {
+  id: string
+  clientSecret: string
+}
+export type PaymentIntentType = 'payment' | 'subscription'
+
+// Make sure to call `loadStripe` outside of a component’s render to avoid
+// recreating the `Stripe` object on every render.
+const stripePromise = loadStripe(
+  'pk_test_51IDGMXEoOT0Jqx0UcoKlkvB7O0VDvFdCBvOCiWiKv6CkSnkZn7IG6cIHuCWg7cegGogYJSy8WsaKzwFHQqN75T7b00d56MtilB'
+)
+
+const cardOptions = {
+  style: {
+    base: {
+      backgroundColor: '#ffffff',
+      color: '#130B43',
+      fontFamily: 'Poppins, sans-serif',
+      fontSmoothing: 'antialiased',
+      fontSize: '16px',
+      lineHeight: '1.5',
+      fontWeight: '500',
+      '::placeholder': {
+        color: '#76709F',
+      },
+    },
+    invalid: {
+      color: '#D03B3B',
+      iconColor: '#D03B3B',
+    },
+  },
+}
+export function StripeForm({
+  paymentIntentType,
+  amountInDollars,
+  onSuccess,
+}: {
+  paymentIntentType: PaymentIntentType
+  onSuccess: () => void
+  amountInDollars: number
+}) {
+  const [succeeded, setSucceeded] = useState(false)
+  const [error, setError] = useState<string | undefined>()
+  const [processing, setProcessing] = useState(false)
+  const [cardValid, setCardValid] = useState(false)
+
+  const createPaymentIntentEndpoint = '/api/v2/donations/payment_intents'
```

Comment:
Is there a Stripe JS library instead of sending bare JSON requests? Might be more reliable to do that.